lib/rbs_rails/active_record.rb

```diff
@@ -371,66 +371,50 @@ def authenticate_#{attribute}: (::String) -> (#{klass_name} | false)
 
       # @rbs singleton: untyped
       private def scopes(singleton:) #: untyped
-        ast = parse_model_file
-        return '' unless ast
-
         prefix = singleton ? 'self.' : ''
 
-        sigs = traverse(ast).map do |node|
-          # @type block: nil | String
-          next unless node.type == :send
-          next unless node.children[0].nil?
-          next unless node.children[1] == :scope
+        scope_definitions = klass.scope_definitions
+        return "" unless scope_definitions
 
-          name_node = node.children[2]
-          next unless name_node
-          next unless name_node.type == :sym
+        enums = klass.enum_definitions.map(&:last).flat_map { [_1, "not_#{_1}"] }
+        sigs = scope_definitions.map do |name, callable|
+          # skip scopes generated by enum
+          next if enums.include?(name.to_s)
 
-          name = name_node.children[0]
-          body_node = node.children[3]
-          next unless body_node
-          next unless body_node.type == :block
```

Comment:
We support scopes with block only so far. But this removes the restriction and allows using scopes with callable objects.

ref: https://craftingruby.com/posts/2015/06/29/query-objects-through-scopes.html